pkgs/node/src/chain.zig

```diff
@@ -461,13 +467,112 @@ pub const BeamChain = struct {
+            self.module_logger.debug("Attestation validation failed: unknown target block root=0x{s}", .{
+                std.fmt.fmtSliceHexLower(&vote.target.root),
+            });
+            return AttestationValidationError.UnknownTargetBlock;
+        };
+
+        const head_idx = self.forkChoice.protoArray.indices.get(vote.head.root) orelse {
+            self.module_logger.debug("Attestation validation failed: unknown head block root=0x{s}", .{
+                std.fmt.fmtSliceHexLower(&vote.head.root),
+            });
+            return AttestationValidationError.UnknownHeadBlock;
+        };
+
+        const source_block = self.forkChoice.protoArray.nodes.items[source_idx];
+        const target_block = self.forkChoice.protoArray.nodes.items[target_idx];
+        const head_block = self.forkChoice.protoArray.nodes.items[head_idx];
+        _ = head_block; // Will be used in future validations
+
+        // 2. Validate slot relationships
+        if (source_block.slot > target_block.slot) {
+            self.module_logger.debug("Attestation validation failed: source slot {d} > target slot {d}", .{
+                source_block.slot,
+                target_block.slot,
+            });
+            return AttestationValidationError.SourceSlotExceedsTarget;
+        }
+
+        //    This corresponds to leanSpec's: assert vote.source.slot <= vote.target.slot
+        if (vote.source.slot > vote.target.slot) {
+            self.module_logger.debug("Attestation validation failed: source checkpoint slot {d} > target checkpoint slot {d}", .{
+                vote.source.slot,
+                vote.target.slot,
+            });
+            return AttestationValidationError.SourceCheckpointExceedsTarget;
+        }
+
+        // 3. Validate checkpoint slots match block slots
+        if (source_block.slot != vote.source.slot) {
+            self.module_logger.debug("Attestation validation failed: source block slot {d} != source checkpoint slot {d}", .{
+                source_block.slot,
+                vote.source.slot,
+            });
+            return AttestationValidationError.SourceCheckpointSlotMismatch;
+        }
+
+        //    This corresponds to leanSpec's: assert target_block.slot == vote.target.slot
+        if (target_block.slot != vote.target.slot) {
+            self.module_logger.debug("Attestation validation failed: target block slot {d} != target checkpoint slot {d}", .{
+                target_block.slot,
+                vote.target.slot,
+            });
+            return AttestationValidationError.TargetCheckpointSlotMismatch;
+        }
+
+        // 4. Validate attestation is not too far in the future
+        //    current_slot = Slot(self.time // SECONDS_PER_INTERVAL)
+        //    assert vote.slot <= Slot(current_slot + Slot(1))
+        const max_allowed_slot = self.forkChoice.fcStore.timeSlots + constants.MAX_FUTURE_SLOT_TOLERANCE;
+        if (vote.slot > max_allowed_slot) {
```

Comment:
need to check this condition, because by this I can attest to next slot for which block has not even been minted, which doesn't sound right